force-app/main/default/classes/DMLOperations.cls

```diff
@@ -146,11 +160,43 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
-    }
+        Account account;
+
+        // find the Account by name; if not found, create a new one
+        try {
+            account = [SELECT Id FROM Account WHERE Name = :accountName LIMIT 1];
+        } 
+        catch (System.QueryException e) {
+            account = new Account(Name = accountName);
+            upsert account; 
+        }
```

Comment:
Usually try and catch blocks are used to throw errors. Try to use an if/else statement